terraform/providers.tf

```diff
@@ -12,4 +12,40 @@ terraform {
 
 provider "azurerm" {
   features {}
-}
\ No newline at end of file
+}
+
+provider "azurerm" {
+  features {}
+  alias           = "stg_peer"
```

Comment:
yeah it is, but you need a provider which has access to both subscriptions/vnets to trigger a peering.

So if I was just to use the provider with no alias to peer ado_vnet TO hub_vnets it would fail because that SP does not have access to the hub vnet/subscription.